### PR TITLE
Kubernetes event source

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @piosz

### CODE REVIEW

File: events/sources/kubernetes/kubernetes_source.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	kubeapi "k8s.io/kubernetes/pkg/api"
+	kubeapiunv "k8s.io/kubernetes/pkg/api/unversioned"
+	kubeclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kubefields "k8s.io/kubernetes/pkg/fields"
+	kubelabels "k8s.io/kubernetes/pkg/labels"
+	kubewatch "k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	// Big enough so it won't be hit anytime soon with resonable GetNewEvents frequency.
```
Comment (gmarek) [inline events/sources/kubernetes/kubernetes_source.go]: Add an info that it's in number of objects (my first thought was bytes).

File: events/sources/kubernetes/kubernetes_source.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	// Large local buffer, periodically read.
+	localEventsBuffer chan *kubeapi.Event
+
+	stopChannel  chan struct{}
+	errorChannel chan error
+
+	eventClient kubeclient.EventInterface
+}
+
+func (this *KubernetesEventSource) GetNewEvents() *core.EventBatch {
```
Comment (gmarek) [inline events/sources/kubernetes/kubernetes_source.go]: I don't see any call for this function. Is there an interface somewhere?

File: events/sources/kubernetes/kubernetes_source.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			this.errorChannel <- fmt.Errorf("Failed to start watch")
+			return
+		}
+
+		watchChannel := watcher.ResultChan()
+		// Inner loop, for update processing.
+		for {
+			select {
+			case watchUpdate, ok := <-watchChannel:
+
```
Comment (gmarek) [inline events/sources/kubernetes/kubernetes_source.go]: remove empty line

File: events/sources/kubernetes/kubernetes_source.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+					if status, ok := watchUpdate.Object.(*kubeapiunv.Status); ok {
+						glog.Errorf("Error during watch: %#v", status)
+						break
+					}
+					glog.Errorf("Received unexpected error: %#v", watchUpdate.Object)
+					break
+				}
+
+				if event, ok := watchUpdate.Object.(*kubeapi.Event); ok {
+
```
Comment (gmarek) [inline events/sources/kubernetes/kubernetes_source.go]: remove empty line

File: events/sources/kubernetes/kubernetes_source.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		}
+	}
+}
+
+func NewKubernetesSource(client kubeclient.Client) *KubernetesEventSource {
+	eventClient := client.Events(kubeapi.NamespaceAll)
+	result := KubernetesEventSource{
+		localEventsBuffer: make(chan *kubeapi.Event, LocalEventsBufferSize),
+		stopChannel:       make(chan struct{}),
+		errorChannel:      make(chan error, 10),
```
Comment (gmarek) [inline events/sources/kubernetes/kubernetes_source.go]: Why 10? It should be either 1, as we always return after seeing an error, or more, but then we need to handle the case when it's full.

File: events/sources/kubernetes/kubernetes_source.go (hunk 1 above)
Comment (mwielgus) [inline events/sources/kubernetes/kubernetes_source.go]: Added.

File: events/sources/kubernetes/kubernetes_source.go (hunk 2 above)
Comment (mwielgus) [inline events/sources/kubernetes/kubernetes_source.go]: Yes. core.EventSource, added comment.

File: events/sources/kubernetes/kubernetes_source.go (hunk 3 above)
Comment (mwielgus) [inline events/sources/kubernetes/kubernetes_source.go]: done

File: events/sources/kubernetes/kubernetes_source.go (hunk 4 above)
Comment (mwielgus) [inline events/sources/kubernetes/kubernetes_source.go]: done

File: events/sources/kubernetes/kubernetes_source.go (hunk 5 above)
Comment (mwielgus) [inline events/sources/kubernetes/kubernetes_source.go]: Ok, 1 is also good. 10 was made just in case someone sometime in the future put more errors there.

**Comment by gmarek** [comment]
LGTM

**Pull request merged**